evm_arithmetization/src/cpu/kernel/asm/memory/syscalls.asm

```diff
@@ -112,6 +112,15 @@ calldataload_large_offset:
 codecopy_within_bounds:
     // stack: total_size, segment, src_ctx, kexit_info, dest_offset, offset, size
     POP
+    // stack: segment, src_ctx, kexit_info, dest_offset, offset, size
```

Comment:
Why do you need to add this changes?

Reply:
`codecopy_within_bounds` calls `codecopy_after` instead of `wcopy_after`. I agree that's more or less code duplication, but I didn't see an easy way to do it more cleanly.

Reply:
From my understanding, the problem with codecopy etc. is that they do not change contexts, but write code in another context. And the idea here is to still be able to prune that extra context when needed by setting the context in codecopy_after.

Reply:
Ahhh, and why `src_ctx` will be never accessed again?

Reply:
In the case of `extcodecopy` and `extcodesize`, we generate `src_ctx` with `next_context_id` which increments `@GLOBAL_METADATA_LARGEST_CONTEXT`, meaning that any other created context will have a different id.

Reply:
Ahhh ok, I think see now. It's because you just need the `new_ctx` for writing the code there (and check the hash etc) but after you copy the code you can forget about it?